core/new-gui/src/app/workspace/component/navigation/navigation.component.ts

```diff
@@ -325,6 +332,56 @@ export class NavigationComponent implements OnInit {
     this.workflowActionService.deleteOperatorsAndLinks(allOperatorIDs, []);
   }
 
+  public onClickImportWorkflow = (file: NzUploadFile): boolean => {
+    const reader = new FileReader();
+    reader.readAsText(file as any);
+    reader.onload = () => {
+      try {
+        const result = reader.result;
+        if (typeof result !== "string") {
+          throw new Error("incorrect format: file is not a string");
+        }
+
+        const workflowContent = JSON.parse(result) as WorkflowContent;
+
+        // set the workflow name using the file name without the extension
+        const fileExtensionIndex = file.name.lastIndexOf(".");
+        var workflowName: string;
+        if (fileExtensionIndex === -1) {
+          workflowName = file.name;
+        } else {
+          workflowName = file.name.substring(0, fileExtensionIndex);
+        }
+        if (workflowName.trim() === "") {
+          workflowName = DEFAULT_WORKFLOW_NAME;
+        }
+
+        const workflow: Workflow = {
+          content: workflowContent,
+          name: workflowName,
+          wid: undefined,
+          creationTime: undefined,
+          lastModifiedTime: undefined,
+        };
+
+        this.workflowActionService.reloadWorkflow(workflow, true);
+      } catch (error) {
+        this.notificationService.error(
+          "An error occured when importing the workflow. Please import a workflow json file."
+        );
+        console.error(error);
+      }
+    };
```

Comment:
I assume we tested import on a recently created empty workflow. but have you tested the following scenarios:
1. import a workflow over an existing workflow
2. undo/redo after import a workflow
3. import a workflow with names that cannot be file name (for example contains `\ /` chars, etc.

Reply:
1. a new workflow is created in this case
2. undo will bring back the overwritten workflow
3. the library we use will automatically convert them as `_`

For the first two behaviors, it just follows logic of `reloadWorkflow`. The `reloadWorkflow` function is supposed to provide certain gurantees to the caller.

Reply:
I don't think reloadworkflow has such guarantees. Can you verify? Reload only clears workspace and redraws the workflow in the workspace, according to my implementation

Reply:
> If this is done with `reloadWorkflow`, then this might conflict with the version preview for now. Can we add a flag to disable the button when `displayParticularWorkflowVersion` is true.

This button is in the group that's hidden when `displayParticularWorkflowVersion` is true, so this button is not there.

Reply:
> I don't think reloadworkflow has such guarantees. Can you verify? Reload only clears workspace and redraws the workflow in the workspace, according to my implementation

I have followed the reload workflow function to control locks and undo/redo stack